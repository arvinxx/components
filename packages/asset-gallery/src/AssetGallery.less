.avx-asset-gallery {
  &-grid {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
  }
  &-masonry {
    column-count: 4;
    column-gap: 0;

    @media (max-width: 400px) {
      column-count: 2;
    }

    @media (max-width: 1200px) {
      column-count: 3;
    }
  }
}
