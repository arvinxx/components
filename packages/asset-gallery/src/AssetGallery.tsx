import type { FC } from 'react';
import React from 'react';
import useMergeValue from 'use-merge-value';

import Layout from './components/Layout';

import AssetCard from './components/AssetCard';

import { YMLToJSON } from './utils/yml';
import type { AssetGalleryProps } from './types';

import './style.less';

const AssetGallery: FC<AssetGalleryProps> = ({
  data,
  darkBackground,
  layout: layoutProps,
  defaultLayout,
  logo,
  showSlider = true,
  onLayoutChange,
  columns: columnsProps,
  defaultColumns,
  onColumnsChange,
  style,
}) => {
  const [columns, setColumn] = useMergeValue(4, {
    value: columnsProps,
    defaultValue: defaultColumns,
    onChange: onColumnsChange,
  });
  const [layout, setLayout] = useMergeValue('masonry', {
    value: layoutProps,
    defaultValue: defaultLayout,
    onChange: onLayoutChange,
  });

  const imageList = typeof data === 'string' ? YMLToJSON(data).data : data;

  return (
    <Layout
      style={style}
      layout={layout}
      onLayoutChange={setLayout}
      logo={logo}
      columns={columns}
      onColumnsChange={setColumn}
      showSlider={showSlider}
    >
      {imageList.map((item, index) => {
        const { description, padding, url, title, dark, sketch } = item;

        const backgroundColor =
          item.darkBackground || darkBackground || '#1890ff';

        return (
          <AssetCard
            key={index}
            padding={padding}
            url={url}
            dark={dark}
            darkBackground={backgroundColor}
            title={title}
            description={description}
            sketch={sketch}
          />
        );
      })}
      <canvas id="canvas" style={{ display: 'none' }} />
    </Layout>
  );
};

export default AssetGallery;
