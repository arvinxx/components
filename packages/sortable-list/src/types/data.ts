import type { UniqueIdentifier } from '@dnd-kit/core';

export interface SortableBaseItem {
  id: UniqueIdentifier;
}

export type SortableItem<T = Record<string, any>> = SortableBaseItem & T;

export type SortableItemList<T = Record<string, any>> = SortableItem<T>[];
