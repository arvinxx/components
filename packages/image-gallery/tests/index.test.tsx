import React from 'react';
import { render } from '@testing-library/react';

import AssetGallery from '@arvinxu/image-gallery';

import { images, pngYML } from './data';

describe('AssetGallery', () => {
  it('默认状态', () => {
    const { container } = render(<AssetGallery data={[]} />);
    expect(container).toMatchSnapshot();
  });

  it('加载数据', () => {
    const { container } = render(<AssetGallery data={images} />);
    expect(container).toMatchSnapshot();
  });

  it('图片源为 PNG', () => {
    const { container } = render(<AssetGallery data={pngYML} />);
    expect(container).toMatchSnapshot();
  });
});
