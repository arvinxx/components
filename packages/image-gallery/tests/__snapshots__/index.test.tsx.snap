// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageGallery 加载数据 1`] = `
<div>
  <div
    class="avx-image-gallery-container avx-image-gallery-masonry"
  >
    <div
      class="avx-image-gallery-item"
    >
      <div
        class="ant-card ant-card-bordered"
      >
        <div
          class="ant-card-cover"
        >
          <div
            class="avx-image-gallery-image-ctn"
            style="border-bottom: 1px solid #f3f3f3;"
          >
            <img
              alt="渐变色 Logo"
              class="avx-image-gallery-image"
              src="https://gw.alipayobjects.com/zos/antfincdn/LFmaI3%26OJh/logo.svg"
            />
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="ant-card-meta"
          >
            <div
              class="ant-card-meta-detail"
            >
              <div
                class="ant-card-meta-title"
              >
                渐变色 Logo
              </div>
              <div
                class="ant-card-meta-description"
              >
                渐变色
              </div>
            </div>
          </div>
        </div>
        <ul
          class="ant-card-actions"
        >
          <li
            style="width: 33.333333333333336%;"
          >
            <span>
              <button
                class="ant-btn ant-btn-link avx-image-gallery-link"
                type="button"
              >
                <span>
                  复制SVG
                </span>
              </button>
            </span>
          </li>
          <li
            style="width: 33.333333333333336%;"
          >
            <span>
              <button
                class="ant-btn ant-btn-link avx-image-gallery-link"
                type="button"
              >
                <span>
                  复制PNG
                </span>
              </button>
            </span>
          </li>
          <li
            style="width: 33.333333333333336%;"
          >
            <span>
              <button
                class="ant-btn ant-btn-link ant-dropdown-trigger avx-image-gallery-link"
                type="button"
              >
                <span
                  aria-label="ellipsis"
                  class="anticon anticon-ellipsis"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="ellipsis"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                    />
                  </svg>
                </span>
              </button>
            </span>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="avx-image-gallery-item"
    >
      <div
        class="ant-card ant-card-bordered"
      >
        <div
          class="ant-card-cover"
        >
          <div
            class="avx-image-gallery-image-ctn"
            style="background: black;"
          >
            <img
              alt="白色 Logo"
              class="avx-image-gallery-image"
              src="https://gw.alipayobjects.com/zos/antfincdn/5qY87lPw9U/white-logo.svg"
              style="padding: 12px;"
            />
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="ant-card-meta"
          >
            <div
              class="ant-card-meta-detail"
            >
              <div
                class="ant-card-meta-title"
              >
                白色 Logo
              </div>
              <div
                class="ant-card-meta-description"
              >
                暗色背景下使用
              </div>
            </div>
          </div>
        </div>
        <ul
          class="ant-card-actions"
        >
          <li
            style="width: 33.333333333333336%;"
          >
            <span>
              <button
                class="ant-btn ant-btn-link avx-image-gallery-link"
                type="button"
              >
                <span>
                  复制SVG
                </span>
              </button>
            </span>
          </li>
          <li
            style="width: 33.333333333333336%;"
          >
            <span>
              <button
                class="ant-btn ant-btn-link avx-image-gallery-link"
                type="button"
              >
                <span>
                  复制PNG
                </span>
              </button>
            </span>
          </li>
          <li
            style="width: 33.333333333333336%;"
          >
            <span>
              <button
                class="ant-btn ant-btn-link ant-dropdown-trigger avx-image-gallery-link"
                type="button"
              >
                <span
                  aria-label="ellipsis"
                  class="anticon anticon-ellipsis"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="ellipsis"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                    />
                  </svg>
                </span>
              </button>
            </span>
          </li>
        </ul>
      </div>
    </div>
    <canvas
      id="canvas"
      style="display: none;"
    />
  </div>
</div>
`;

exports[`ImageGallery 默认状态 1`] = `
<div>
  <div
    class="avx-image-gallery-container avx-image-gallery-masonry"
  >
    <canvas
      id="canvas"
      style="display: none;"
    />
  </div>
</div>
`;
